Movies_mvp/Movies_mvp/MovieCatalogViewController/Presenter/MovieCatalogPresenter.swift

```diff
@@ -0,0 +1,80 @@
+// MovieCatalogPresenter.swift
+// Copyright © RoadMap. All rights reserved.
+
+import Foundation
+
+/// Презентер каталога фильмов
+final class MovieCatalogPresenter: MovieCatalogPresenterProtocol {
+    // MARK: - Public Properties
+
+    weak var view: MovieCatalogViewProtocol?
+    var router: RouterProtocol?
+    var movies: [Movie] = []
+
+    // MARK: - Private Properties
+
+    private let networkService: NetworkServiceProtocol
+    private var hasNextPage = true
+    private var currentPage = 1
+    private var currentRequestType: RequestType = .popular
+
+    // MARK: - Initializers
+
+    init(view: MovieCatalogViewProtocol?, networkService: NetworkServiceProtocol, router: RouterProtocol) {
+        self.view = view
+        self.networkService = networkService
+        self.router = router
+    }
+
+    // MARK: - Public methods
+
+    func fetchResult(requestType: RequestType) {
```

Comment:
```suggestion
    func fetchResult(requestType: RequestType) {
```
fetchResult ? ЛУчше запрос фильмов чем результата